Client/trunk/ParaEngineClient/BlockEngine/BlockWorldClient.cpp

```diff
@@ -3235,7 +3235,12 @@ namespace ParaEngine
 			auto light_ambient = light_param->Ambient;
 
 			auto light_position = light_param->Position;
-			auto light_direction = light_param->GetDirection();
+
+			if (lightObject->IsRotationDirty()) {
+				light_param->RecalculateDirection();
+				lightObject->SetRotationDirty(false);
+			}
```

Comment:
better refactor these code to light object's GetDirection() method

Reply:
👍   please review again